sklearn/tests/test_pipeline.py

```diff
@@ -983,6 +983,72 @@ def test_set_feature_union_step_drop():
     assert not record
 
 
+def test_set_feature_union_passthrough():
+    mult2 = Mult(2)
+    mult2.get_feature_names = lambda: ["x2"]
+    mult3 = Mult(3)
+    mult3.get_feature_names = lambda: ["x3"]
+    X = np.asarray([[1]])
+
+    ft = FeatureUnion([("m2", mult2), ("m3", mult3)])
+    assert_array_equal([[2, 3]], ft.fit(X).transform(X))
+    assert_array_equal([[2, 3]], ft.fit_transform(X))
+
+    with pytest.warns(None) as record:
```

Comment:
I am just clarifying if I properly understood what you're asking for here. Shall I use various classes of warning in `pytest.warns` instead of `pytest.warns(None)`?

Reply:
No, you can just remove the full context manager.

Reply:
I don't get it. If we remove the context manager then the test will not fail, it will just issue a warning that will be ignored, no?

Reply:
I did a git blame of the previous test to understand why we wanted to be sure that no warning would be raised. The reason is that in the past we could use `None` and `"drop"` to do the same and `None` got deprecated. @thomasjpfan implemented a safeguard to be sure that no deprecation warning was raised in the case of `"drop"` (thought this case would have been watched by our `-Werror::FutureWarning` in the CI anyway.

So here I think that we don't need to make the test more complex with the decorator and just remove it.

Reply:
I removed the context managers.